apps/arena_load_test/lib/arena_load_test/socket_supervisor.ex

```diff
@@ -1,50 +1,78 @@
 defmodule ArenaLoadTest.SocketSupervisor do
   @moduledoc """
-  Socket Supervisor
+  Dynamic Supervisor for the websockets connections.
   """
   use DynamicSupervisor
   alias ArenaLoadTest.SocketHandler
   alias ArenaLoadTest.GameSocketHandler
+  alias ArenaLoadTest.LoadtestManager
   require Logger
 
   def start_link(args) do
     DynamicSupervisor.start_link(__MODULE__, args, name: __MODULE__, max_restarts: 1)
   end
 
+  @impl true
+  def init(_opts) do
+    DynamicSupervisor.init(strategy: :one_for_one)
+  end
+
+  @doc """
+  Initializes a websocket that handles the client connection in the game waiting queue.
+  """
   def add_new_client(client_id) do
+    true = :ets.insert(:clients, {client_id, client_id})
+
     DynamicSupervisor.start_child(
       __MODULE__,
       {SocketHandler, client_id}
     )
   end
 
+  @doc """
+  Initializes a websocket that handles the client connection in-game.
+  """
   def add_new_player(client_id, game_id) do
     DynamicSupervisor.start_child(
       __MODULE__,
       {GameSocketHandler, {client_id, game_id}}
     )
   end
 
-  @impl true
-  def init(_opts) do
-    DynamicSupervisor.init(strategy: :one_for_one)
-  end
+  @doc """
+  Loadtests entrypoint.
+  Creates given amount of clients that will join and play a game for the given duration.
+  """
+  def spawn_players(num_clients, playtime_duration_ms \\ 999_999) do
+    send(LoadtestManager, :clients_log)
+    Process.send_after(LoadtestManager, :loadtest_finished, playtime_duration_ms)
 
-  # Creates `num_clients` clients to join a game
-  def spawn_players(num_clients) do
-    case :ets.whereis(:clients) do
-      :undefined -> :ets.new(:clients, [:set, :named_table, :public])
-      _table_exists_already -> nil
-    end
+    create_ets_table(:clients)
```

Comment:
Common place for doing this is in the `init/1` function

Reply:
Alright. Done!